Sources/AuthFoundation/User Management/Credential.swift

```diff
@@ -229,12 +229,10 @@ public final class Credential: Equatable, OAuth2ClientDelegate {
         let shouldRemove = shouldRemove(for: type)
         
         oauth2.revoke(token, type: type) { result in
-            defer { completion?(result) }
-            
-            guard case .success = result else { return }
-            
             // Remove the credential from storage if the access token was revoked
-            if shouldRemove {
+            if case .success = result,
```

Comment:
@baksha97 If the revoke is unsuccessful, the developer could still call `remove()`.  However, if the revoke operation failed due to a network timeout/offline error, the token will still be valid on the server, and removing it will prevent the app from being able to revoke it in the future.

For example, you could do the following:

```
do {
  try await credential.revoke()
} catch {
  // Check to see if the error is recoverable
  if shouldRemoveTheToken {
    try credential.remove()
  }
}
```